File: shopping-cart-frontend/src/services/cartService.js
import apiClient from "./apiClient";

const cartService = {

    addToCart: async (

        productId,

        quantity

    ) => {

        const response = await apiClient.post(

            "/cart/add",

            {

                productId,

                quantity

            }

        );

        return response.data;

    },


    getCart: async () => {

        const response = await apiClient.get(

            "/cart"

        );

        return response.data;

    },


    updateCartItem: async (

        cartItemId,

        quantity

    ) => {

        const response = await apiClient.put(

            `/cart/items/${cartItemId}`,

            {

                quantity

            }

        );

        return response.data;

    },


    removeCartItem: async (

        cartItemId

    ) => {

        await apiClient.delete(

            `/cart/items/${cartItemId}`

        );

    },


    clearCart: async () => {

        await apiClient.delete(

            "/cart"

        );

    }

};


export default cartService;

File: shopping-cart-backend/src/main/java/com/basil/shoppingcart/service/CartService.java
package com.basil.shoppingcart.service;

import com.basil.shoppingcart.dto.response.CartResponse;

public interface CartService {

    CartResponse addProductToCart(Long productId, Integer quantity);

    CartResponse getMyCart();

    CartResponse updateCartItem(Long cartItemId, Integer quantity);

    void removeCartItem(Long cartItemId);

    void clearCart();

}